test/feature_check_test.cpp

```diff
@@ -0,0 +1,190 @@
+
+    // 19: Shader Cache
+    D3D12_SHADER_CACHE_SUPPORT_FLAGS ShaderCacheSupportFlags = features->ShaderCacheSupportFlags();
+
+    // 20: Command Queue Prioirity
+    BOOL CommandQueuePrioritySupportedPositive = features->CommandQueuePrioritySupported(D3D12_COMMAND_LIST_TYPE_DIRECT, 0);
+    BOOL CommandQueuePrioritySupportedNegative = features->CommandQueuePrioritySupported(D3D12_COMMAND_LIST_TYPE_COPY, D3D12_COMMAND_QUEUE_PRIORITY_GLOBAL_REALTIME);
+    BOOL CommandQueuePrioritySupportedInvalid = features->CommandQueuePrioritySupported((D3D12_COMMAND_LIST_TYPE)7, 0);
+    
+    // 21: Options3
+    BOOL CopyQueueTimestampQueriesSupported = features->CopyQueueTimestampQueriesSupported();
+    BOOL CastingFullyTypedFormatSupported = features->CastingFullyTypedFormatSupported();
+    D3D12_COMMAND_LIST_SUPPORT_FLAGS WriteBufferImmediateSupportFlags = features->WriteBufferImmediateSupportFlags();
+    D3D12_VIEW_INSTANCING_TIER ViewInstancingTier = features->ViewInstancingTier();
+    BOOL BarycentricsSupported = features->BarycentricsSupported();
+
+    // 22: Existing Heaps
+    BOOL ExistingHeapsSupported = features->ExistingHeapsSupported();
+
+    // 23: D3D12 Options4
+    BOOL MSAA64KBAlgnedTextureSupported = features->MSAA64KBAlignedTextureSupported();
+    D3D12_SHARED_RESOURCE_COMPATIBILITY_TIER SharedResourceCompatibilityTier = features->SharedResourceCompatibilityTier();
+    BOOL Native16BitShaderOpsSupported = features->Native16BitShaderOpsSupported();
+
+    // 24: Serialization
+    D3D12_HEAP_SERIALIZATION_TIER HeapSerializationTier = features->HeapSerializationTier();
+
+    // 25: Cross Node
+    //  D3D12_CROSS_NODE_SHARING_TIER CrossNodeSharingTier = features->CrossNodeSharingTier(); // Same option in D3D12Options
+    BOOL CrossNodeAtomicShaderInstructions = features->CrossNodeAtomicShaderInstructions();
+
+    // 27: Options5
+    BOOL SRVOnlyTiledResourceTier3 = features->SRVOnlyTiledResourceTier3();
+    D3D12_RENDER_PASS_TIER RenderPassesTier = features->RenderPassesTier();
+    D3D12_RAYTRACING_TIER RaytracingTier = features->RaytracingTier();
+
+    // 28: Displayable
+    BOOL DisplayableTexture = features->DisplayableTexture();
+    D3D12_SHARED_RESOURCE_COMPATIBILITY_TIER DisplayableSharedResourceCompatibilityTier = features->DisplayableSharedResourceCompatibilityTier();
+
+    // 30: Options6
+    BOOL AdditionalShadingRatesSupported = features->AdditionalShadingRatesSupported();
+    BOOL PerPrimitiveShadingRateSupportedWithViewportIndexing = features->PerPrimitiveShadingRateSupportedWithViewportIndexing();
+    D3D12_VARIABLE_SHADING_RATE_TIER VariableShadingRateTier = features->VariableShadingRateTier();
+    BOOL ShadingRateImageTileSize = features->ShadingRateImageTileSize();
+    BOOL BackgroundProcessingSupported = features->BackgroundProcessingSupported();
+
+    // 31: Query Metacommand
+    // TODO: Don't know how to test
```

Comment:
Query metacommand requires a GUID as an input parameter and I don't know how to use it :( Does someone know how to test this feature check?

Reply:
The new API basically makes a forwarding call to the original API, so there's not much to test to ensure its correctness besides making sure it passes the params correctly. I'm going to skip it for now but I will write some basic test cases in the test suite.

Reply:
Makes sense!